docs/_writing-tests/testdriver.md

```diff
@@ -48,4 +48,34 @@ between the function being called and the promise settling.
 To send special keys, one must send the respective key's codepoint. Since this uses the WebDriver protocol, you can find a [list for code points to special keys in the spec](https://w3c.github.io/webdriver/webdriver-spec.html#keyboard-actions).
 For example, to send the tab key you would send "\uE004".
 
+### `test_driver.pointer_action_sequence(actions)`
+ - `actions` <[Array]<[Object]>> an array of Action objects`
+  - `action` <[Object]> A single action
+   - `id` The ID of the pointer to use. If one with this id does not exist a new one will be used
+   - `pointerType` The type of pointer to use, an enum of `mouse`, `touch` and `pen`.
+   - `subtype` The type of action to perform, an enum of `pointerUp`, `pointerMove`, `pointerDown` and `pause`
+   - `target` Either a DOM Element object or an Object with exactly two integer attrbutes `x` and `y`.
+   - `duration` The duration in ms before the next action in the sequence should start
+
+
+This function causes a sequence of pointer actions to be sent to the browser. It returns a `Promise` that
+resolves after the actions have been sent or rejects if an error was thrown.
+
+### `test_driver.send_keys(element, keys)`
+#### `element: a DOM Element object`
+#### `keys: string to send to the element`
+
+This function causes the string `keys` to be send to the target
```

Comment:
Will you call .focus() programmatically in js to get the target element focused? If not how do you want the browser to send the event to that element? If not maybe remove the element altogether and then browser decides the given key will to which element (which will be the focused element).

Reply:
I'm not sure I have context on what focus does, and I am also confused with the line you are referring to.

Reply:
I'm referring to this API as whole and how it is implemented. Although in this PR you only added this documentation but the SendKeysAction doesn't seem to call focus on the given element. So I was wondering how it would possibly route a key event to a particular given element as explained in this paragraph.

Reply:
I think this is a strange git diff error since I didn't change this part of the doc but only seem to have moved it. I will try and update my PR so it doesn't think I moved it.
Can you open up an issue for that and tag me in it?